File: No_2053/Kth Distinct String in an Array.cpp
class Solution {
public:
    string kthDistinct(vector<string>& arr, int k) {
        unordered_map<string, int> count;
        for(int i=0; i<arr.size(); i++) {
            count[arr[i]]++;
        }
        for(int i=0; i<arr.size(); i++) {
            if(count[arr[i]] == 1) {
                if(--k == 0) {
                    return arr[i];
                }
            }
        }
        
        return "";
        
    }
};

File: No_2053/Kth Distinct String in an Array.py
class Solution {
public:
    string kthDistinct(vector<string>& arr, int k) {
        unordered_map<string, int> counter;
        for (string s : arr)
            counter[s]++;
        
        for (string s : arr) {
            if (counter[s] == 1) {
                if (k == 1)
                    return s;
                k--;
            }
        }
        
        return "";
    }
};